test/hotspot/jtreg/vmTestbase/nsk/jdi/EventSet/suspendPolicy/suspendpolicy008.java

```diff
@@ -418,8 +419,6 @@ private ThreadStartRequest settingThreadStartRequest(int suspendPolicy,
             throws JDITestRuntimeException {
         try {
             ThreadStartRequest tsr = eventRManager.createThreadStartRequest();
-//            tsr.addThreadFilter(mainThread);
```

Comment:
I suppose the fix might be just to uncomment this line.
Your solution is more complex, but should work too

Reply:
I don't think that filter is correct. ThreadStartEvents are always delivered on the thread that is starting. Adding a thread filter for "main" is going to result in never finding a matching ThreadStartEvent since "main" has already started and it won't match any of the newly started threads.

Reply:
Oh, yeah. You are right.